radis/api/exomolapi.py

```diff
@@ -1565,18 +1609,45 @@ def download(self, molec, extension, numtag=None):
         tag = molec.split("__")
         molname_simple = exact_molname_exomol_to_simple_molname(tag[0])
         # TODO: add progress bar
+        total_download_size_gb = 0
         for ext in extension:
             if ext == ".trans.bz2" and numtag is not None:
                 ext = "__" + numtag + ext
 
             if ext == ".broad":
-                partners_success = np.ones(len(self.broad_partners), dtype=bool)
+                pfname_arr, url, partners_success = self._construct_filenames_and_url(
+                    ext, molname_simple, tag, molec, numtag
+                )
+            else:
+                pfname_arr, url = self._construct_filenames_and_url(
+                    ext, molname_simple, tag, molec, numtag
+                )
+
+            total_download_size_gb += self._calculate_download_size(url, pfname_arr)
+
+        print(f"Total download size: {total_download_size_gb:.6f} GB")
+        MAX_SIZE_GB = config.get("PREVENT_LARGE_DOWNLOAD_ABOVE_X_GB", 1)
 
-                pfname_arr = [tag[0] + "__" + s + ext for s in self.broad_partners]
-                url = EXOMOL_URL + molname_simple + "/" + tag[0] + "/"
+        if total_download_size_gb > MAX_SIZE_GB:
+            warning_msg = (
+                f"The total download size is {total_download_size_gb:.2f} GB, "
+                f"which exceeds the threshold of {MAX_SIZE_GB} GB. "
+                "Downloading may consume significant resources and take time."
```

Comment:
```suggestion
            warning_msg = (
                f"The total download size is {total_download_size_gb:.2f} GB, which will take time and potential a significant portion of your disk memory."
                "To prevent this warning, you increase the limit using `radis.config["PREVENT_LARGE_DOWNLOAD_ABOVE_X_GB"] =  1`."
```
The message "exceeds the threshold" might be understood as "it doesn't work". Please consider rephrasing.

Reply:
Yes, that make sense! Updated the warning message.